frontend/canonical.php

```diff
@@ -252,7 +252,7 @@ protected function redirect_canonical( $url, $language ) {
 
 		$backup_wp_query = $wp_query;
 
-		if ( isset( $wp_query->tax_query ) ) {
+		if ( ! is_null( $wp_query->tax_query ) ) {
```

Comment:
This rather looks like a trick reliying on a wrong PHPStan detection. The issue is not on `tax_query` but on `query`
1. Currently `$wp_query` is an anonymous object for PHPStan, so we should check that the property exists before accessing it.
```php
if ( isset( $wp_query->query ) ) {
	unset( $wp_query->query['lang'] );
}
```
This is one solution.
2. A second solution would be to document the variable as `WP_Query` as you mentioned. The new reported error is due to a doc error in WP. You could open a ticket and meanwhile we could temporarily ignore this error. 
3. Side note: Don't use `is_null()` but `null ===` instead which is faster (speed is not important here but it's a good habit to have).

Reply:
I chose second solution in 7529b53faa2c646b1912fb09f55c3728ba2a0c6e 
See https://core.trac.wordpress.org/ticket/60563